lib/core/proxy.c

```diff
@@ -66,7 +66,7 @@ static h2o_iovec_t rewrite_location(h2o_mem_pool_t *pool, const char *location,
         goto NoRewrite;
     if (loc_parsed.scheme != &H2O_URL_SCHEME_HTTP)
         goto NoRewrite;
-    if (!h2o_lcstris(loc_parsed.host.base, loc_parsed.host.len, match->host.base, match->host.len))
+    if (!h2o_url_compare_hosts(loc_parsed.host, match->host, match->host_is_unix_path))
```

Comment:
Shouldn't the logic be:
* the result is a mismatch if either is a unix path but the other is not
* if both are unix paths, then perform case-sensitive comparison
* if both are not unix paths, then perform case-insensitive comparison